podaac/merger/merge_worker.py

```diff
@@ -25,6 +26,36 @@ def shared_memory_size():
         return int(default_memory_size)
 
 
+def max_var_memory(file_list, var_info, max_dims):
+    """
+    function to get the maximum shared memory that will be used for variables
+
+    Parameters
+    ----------
+    file_list : list
+        List of file paths to be processed
+    var_info : dict
+        Dictionary of variable paths and associated VariableInfo
+    """
+
+    max_var_mem = 0
+    for file in file_list:
+        with nc.Dataset(file, 'r') as origin_dataset:
+
+            for var_path, var_meta in var_info.items():
+                ds_group, var_name = resolve_group(origin_dataset, var_path)
+                ds_var = ds_group.variables.get(var_name)
+                if ds_var is None:
+                    target_shape = tuple(max_dims[f'/{dim}'] for dim in var_meta.dim_order)
+                    var_size = math.prod(target_shape) * var_meta.fill_value
```

Comment:
Just a quick note - I think you’re missing a `.nbytes` here like you used on line 53

Reply:
fixed changed to use var_meta.datatype.itemsize